### ISSUE TITLE
Write EFA() function for lavaan - Reminder

### ISSUE THREAD

**Issue by hugoharada** [yrosseel/lavaan#112]
Write efa() function to address issues discussed in the following thread.

https://groups.google.com/forum/#!topic/lavaan/ujkHmCVirEY

**Comment by dugaldfoster** [comment]
Hi Hugo/Yves, just wondering if you know whether there's been any progress on this?

**Comment by fabriciofialho** [comment]
Hi,

semTools has the functions efaUnrotate(), orthRotate(), and oblqRotate()
which allow to run EFA (first command) and to rotate the solution (latter
two commands). They have worked greatly so far. Have you experienced issues
with those functions?

**Comment by dugaldfoster** [comment]
Hi Fabricio,

Thanks for your reply. I was just wondering whether lavaan also offered EFA options, or if the idea to add that had been dropped. In the thread, Yves mentioned a way to "hack" an EFA in lavaan using the std.lv = and orthogonal = TRUE specifications. Would this just give the same results as the equivalent function in semTools?

**Comment by yrosseel** [comment]
Full support for EFA has been included in 0.6-4. There is no dedicated function yet, but the following syntax can be used the with sem/lavaan functions:

model <-  '
    efa("efa")*f1 + 
    efa("efa")*f2 + 
    efa("efa")*f3 + 
    efa("efa")*f4 =~ y1 + y2 + y3 + y4 + y5 + y6 +
                     y7 + y8 + y9 + y10 + y11 + y12
'

fit <- sem(model, data = Data, rotation = "geomin", 
                rotation.args = list(geomin.epsilon = 0.01, rstarts = 30,
                                             algorithm = "gpa", orthogonal = FALSE,
                                             row.weights = "none", std.ov = TRUE))

Documentation is on its way.

**Comment by mrli1234** [comment]
> 
> 
> Full support for EFA has been included in 0.6-4. There is no dedicated function yet, but the following syntax can be used the with sem/lavaan functions:
> 
> model <- '
> efa("efa")*f1 +
> efa("efa")*f2 +
> efa("efa")*f3 +
> efa("efa")*f4 =~ y1 + y2 + y3 + y4 + y5 + y6 +
> y7 + y8 + y9 + y10 + y11 + y12
> '
> 
> fit <- sem(model, data = Data, rotation = "geomin",
> rotation.args = list(geomin.epsilon = 0.01, rstarts = 30,
> algorithm = "gpa", orthogonal = FALSE,
> row.weights = "none", std.ov = TRUE))
> 
> Documentation is on its way.

Thank you very much for the lavaan package and the syntax.  Would you please let me know how to add a bi-factor the model?  Thanks and  happy new year!

**Comment by TDJorgensen** [comment]
> how to add a bi-factor the model? 

A bifactor model is a CFA model.  You just need to specify your hypothesized factors in your model syntax.  For example, adding a general factor to the `?cfa` help-page example:

```
HS.model <- ' visual  =~ x1 + x2 + x3
              textual =~ x4 + x5 + x6
              speed   =~ x7 + x8 + x9 
G =~ x1 + x2 + x3 + x4 + x5 + x6 + x7 + x8 + x9 '

fit <- cfa(HS.model, data = HolzingerSwineford1939, 
           orthogonal = TRUE, std.lv = TRUE)
```

**Comment by mrli1234** [comment]
> 
> 
> > how to add a bi-factor the model?
> 
> A bifactor model is a CFA model. You just need to specify your hypothesized factors in your model syntax. For example, adding a general factor to the `?cfa` help-page example:
> 
> ```
> HS.model <- ' visual  =~ x1 + x2 + x3
>               textual =~ x4 + x5 + x6
>               speed   =~ x7 + x8 + x9 
> G =~ x1 + x2 + x3 + x4 + x5 + x6 + x7 + x8 + x9 '
> 
> fit <- cfa(HS.model, data = HolzingerSwineford1939, 
>            orthogonal = TRUE, std.lv = TRUE)
> ```

Thank you very much for your reply.  This is a normal bifactor model.

However, I think the model provided by  yrosseel  is an exploratory sem model(esem)? I would to establish a bifactor-esem model.

Thanks again and happy new year!

**Comment by yrosseel** [comment]
Bifactor rotation is on my TODO list for 0.6-9

**Comment by mrli1234** [comment]
> 
> 
> Bifactor rotation is on my TODO list for 0.6-9

Thanks. Hope to see it soon.

**Comment by bwiernik** [comment]
@yrosseel Which bifactor rotations are you intending to implement? The Jennrich rotations appear to perform poorly, but a direct Schmid-Leiman rotation based on a target matrix appears to work well https://psycnet.apa.org/record/2019-42954-001

**Comment by cmerinos** [comment]
> 
> 
> Full support for EFA has been included in 0.6-4. There is no dedicated function yet, but the following syntax can be used the with sem/lavaan functions:
> 
> model <- '
> efa("efa")*f1 +
> efa("efa")*f2 +
> efa("efa")*f3 +
> efa("efa")*f4 =~ y1 + y2 + y3 + y4 + y5 + y6 +
> y7 + y8 + y9 + y10 + y11 + y12
> '
> 
> fit <- sem(model, data = Data, rotation = "geomin",
> rotation.args = list(geomin.epsilon = 0.01, rstarts = 30,
> algorithm = "gpa", orthogonal = FALSE,
> row.weights = "none", std.ov = TRUE))
> 
> Documentation is on its way.

Dear @yrosseel , greetings.
A question: in this script, one can use target rotation? Or not yet? Thanks
